cns/cniconflist/generator.go

```diff
@@ -0,0 +1,40 @@
+package cniconflist
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/Azure/azure-container-networking/cns/configuration"
+)
+
+type Generator struct {
+	writer   io.WriteCloser
+	scenario configuration.CNIConflistScenario
+}
+
+func (g *Generator) Close() error {
+	if err := g.writer.Close(); err != nil {
+		return fmt.Errorf("error closing generator: %w", err)
+	}
+
+	return nil
+}
+
+// cniConflist represents the containernetworking/cni/pkg/types.NetConfList
+//
+// TODO: should use the containernetworking/cni/pkg/types.NetConfList, but our CNI config
+// doesn't embed the NetConf struct, so we need to refactor that before we can use the
+// upstream containernetworking types
+type CNIConflist struct {
+	CNIVersion   string        `json:"cniVersion,omitempty"`
+	Name         string        `json:"name,omitempty"`
+	DisableCheck bool          `json:"disableCheck,omitempty"`
+	Plugins      []interface{} `json:"plugins,omitempty"`
```

Comment:
Not necessary for this PR, but I've been trying to force myself to use `any` over `interface{}` now that we can. It's a hard habit to break.

Reply:
done